controllers/infinispan_controller.go

```diff
@@ -1280,7 +1281,8 @@ func (r *infinispanRequest) statefulSetForInfinispan(adminSecret, userSecret, ke
 		pvc.OwnerReferences[0].BlockOwnerDeletion = pointer.BoolPtr(false)
 		// Set a storage class if it specified
 		if storageClassName := ispn.StorageClassName(); storageClassName != "" {
-			if _, err := kube.FindStorageClass(storageClassName, r.Client, r.ctx); err != nil {
+			storageClass := &storagev1.StorageClass{}
+			if _,err := kube.LookupResource(storageClassName, ispn.Namespace, storageClass, ispn, r.Client, reqLogger, r.eventRec, r.ctx); err != nil {
```

Comment:
It wasn't the original aim of this PR, but I think we can remove this lookup altogether. If the StorageClass doesn't exist, then the StatefulSet creation will fail.

This is better from a user perspective, as a user who is able to create a Infinispan CR will also be able to inspect created resources in their namespace. However, it's not guaranteed that they will have access to the operator log if it is installed cluster wide and they will not know why the Infinispan cluster cannot be created.

Reply:
@leo8a On second thought I think we should keep the `LookupResource` call. @dmvolod Makes an excellent point that the user will be notified via the event logger that the `StorageClass` is not available, so we can avoid trying to create the StatefulSet with an invalid config and the user will be notified.

Reply:
agree.